### PR TITLE
Simplify waf usage

### PR DESCRIPTION
**Pull request by mortenvp**
Added a few import and loads to simplify the usage our custom waf, a wscript can now be:

```
#! /usr/bin/env python
# encoding: utf-8

top = '.'

def resolve(ctx):

    ctx.add_git_semver_dependency(
        name='waf-tools',
        git_repository='github.com/steinwurf/waf-tools.git',
        major=3)

def configure(conf):
    pass

def build(bld):
    pass
```

### CODE REVIEW

File: tools/wurf_options.py [hunk 1]
```diff
@@ -63,6 +66,20 @@ def resolve(ctx):
...
 class WurfOptions(Options.OptionsContext):
 
+    def execute(self):
+
+        # Execute the context, this will go out and invoke the option(...)
+        # function defined in the wscript. We have some extra tools that
+        # also define the option(...) function to avoid having to call load
+        # in all wscripts to initialize those, we can do it here.
+
+        self.load('wurf_common_tools')
```
Comment (petya2164) [inline tools/wurf_options.py]: What happens if I don't want to load all the common tools in some custom project? (perhaps  a full python project would not need the common tools)

Comment (petya2164) [inline tools/wurf_options.py]: Furthermore, can we assume that wurf_common_tools will be loaded automagically in the configure and build steps? (it has to be loaded before anything else is executed in those functions)

Comment (petya2164) [inline tools/wurf_options.py]: I think the answer is no to my last question. Your changes only hide the load operation for options and resolve. It just feels weird that you have to load the tool manually in some steps, but in some other steps it is loaded for you automagically. I would rather use the explicit load operation at all places!
The explicit solution would also allow us to *not* load all the common tools where we don't need them.

File: tools/wurf_dependency_bundle.py [hunk 2]
```diff
@@ -99,6 +99,34 @@ def add_dependency(ctx, resolver, recursive_resolve=True, optional=False):
             if recursive_resolve:
                 ctx.recurse([path])
 
+@conf
+def add_git_semver_dependency(ctx, name, git_repository, major, minor=None,
+                              patch=None, recursive_resolve=True,
+                              optional=False):
+    """
+
```
Comment (jpihl) [inline tools/wurf_dependency_bundle.py]: remove newline

File: tools/wurf_dependency_bundle.py [hunk 3]
```diff
@@ -99,6 +99,34 @@ def add_dependency(ctx, resolver, recursive_resolve=True, optional=False):
...
+    Adds ResolveVersion dependency (this function exists only such that
+    wscripts do not have to perform an import to access the ResolveVersion
+    class)..
```
Comment (jpihl) [inline tools/wurf_dependency_bundle.py]: https://www.python.org/dev/peps/pep-0257/

File: tools/wurf_options.py (hunk 1 above)
Comment (mortenvp) [inline tools/wurf_options.py]: To the first, if you don't want to load the tools, then probably you should use a different waf (e.g. the stock one). I think that would be fine. 

It is easy to add such that they will be loaded in both Options, Resolve, Configuration and Build contexts. That is probably a good idea (if needed I will check that).

Comment (petya2164) [inline tools/wurf_options.py]: You are right, maybe we can also control the loaded tools with some environment variables (if we get a situation where a particular tool is incompatible with a project). But let's cross that bridge when we get there.

I'm not sure if hiding the load operation is a good idea in any step, but you can experiment with that solution before this change is merged :)
Currently, we have a great explicit solution and it is easy to follow what is loaded where. We don't gain much by removing those 4 lines, and we definitely make things more obscure!

Comment (mortenvp) [inline tools/wurf_options.py]: Roger that. I will experiment a bit with the implicit loading. Functionally these changes should not make any real difference - in the sense you can choose to remove those lines. However, you can also keep them, our existing code should still work fine with this version.

Comment (petya2164) [inline tools/wurf_options.py]: Are you sure about that last statement? 
If you try to use this version with our existing wscripts, then wurf_common_tools will be loaded twice (in some steps). That might have some unintended side effects. We need to understand the loading mechanism in waf to ascertain those side effects.
I am also trying to make a general point here: "Explicit is better than implicit" ;)

File: tools/wurf_options_context.py [hunk 4]
```diff
@@ -7,6 +7,9 @@
 
     import waflib.extras.wurf_options
 
+This step can be done automatically by building a custom waf binary and
+using the --prelude statement (see more here http://bit.ly/1Te4mRz).
+
```
Comment (petya2164) [inline tools/wurf_options_context.py]: Unfortunately, the prelude magic fails with the following error:
```
[localhost] local: python waf-light --make-waf --tools=compat15,C:\steinwurf\waf\tools\wurf_common_tools.py,C:\steinwurf\waf\tools\wurf_configure_outp
ut.py,C:\steinwurf\waf\tools\wurf_dependency_bundle.py,C:\steinwurf\waf\tools\wurf_dependency_resolve.py,C:\steinwurf\waf\tools\wurf_git.py,C:\steinwu
rf\waf\tools\wurf_options.py,C:\steinwurf\waf\tools\wurf_resolve_context.py,C:\steinwurf\waf\tools\wurf_standalone.py,C:\steinwurf\waf\python-semver\s
emver.py --prelude='    import waflib.extras.wurf_options'
------> Executing code from the top-level wscript <-----
No function import defined in C:\steinwurf\waf\waf\wscript
```

For the time being, I reverted the corresponding change in fabric. We can take a look at it later.

Comment (mortenvp) [inline tools/wurf_options_context.py]: Ok, yes did not test it on Windows.. probably some issue there then

**Pull request merged**